core_lib/src/structure/filemanager.cpp

```diff
@@ -562,7 +562,7 @@ int FileManager::countExistingBackups(const QString& fileName) const
     const QString& baseName = fileInfo.completeBaseName();
 
     int backupCount = 0;
-    for (QFileInfo dirFileInfo : directory.entryInfoList(QDir::Filter::Files)) {
+    for (const QFileInfo &dirFileInfo : directory.entryInfoList(QDir::Filter::Files)) {
```

Comment:
```suggestion
    for (const QFileInfo& dirFileInfo : qAsConst(directory.entryInfoList(QDir::Filter::Files))) { {
```

By adding const here, you'll get an additional warning saying 'Might detach container', as such we use qAsConst(...).

Reply:
adding it like that fails:
![image](https://github.com/user-attachments/assets/57c599ae-5754-4173-8a37-755cd162dbc3)

adding it just around `directory` doesn't make the warning go away:
![image](https://github.com/user-attachments/assets/2ce796f7-8519-4f62-a404-7dc101918ab4)